### PR TITLE
[ICDS Dashboard]: PPR frontend Issues

### PR DESCRIPTION
**Pull request by ShubhamBansal1997**
##### SUMMARY
Disabling the export option if the view by is not selected
Handling the quarter disruption if the Date Period is selected.

### CODE REVIEW

File: custom/icds_reports/templates/icds_reports/icds_app/download.directive.html [hunk 1]
```diff
@@ -119,7 +119,7 @@
                            popover-placement="top-right"
                            popover-trigger="'mouseenter'"></i></label>
                     <div class="col-sm-6">
-                        <ui-select ng-model="$ctrl.selectedDataPeriod" theme="bootstrap">
+                        <ui-select ng-model="$ctrl.selectedDataPeriod" theme="bootstrap" on-select="$ctrl.onDataPeriodSelect($item)">
```
Comment (Rohit25negi) [inline custom/icds_reports/templates/icds_reports/icds_app/download.directive.html:122]: why do you need both here? The Model as well and on-select callback as well?

File: custom/icds_reports/static/js/directives/download/download.directive.js [hunk 2]
```diff
@@ -360,7 +360,7 @@ function DownloadController($scope, $rootScope, $location, locationHierarchy, lo
             vm.setPPRYears();
 
         } else if (year.id === latest.getFullYear()) {
-            const maxQuarter = Math.floor((latest.getMonth() + 1)/4);
+            var maxQuarter = Math.floor((latest.getMonth() + 1) / 3);
```
Comment (Rohit25negi) [inline custom/icds_reports/static/js/directives/download/download.directive.js:363]: which Issues its addressing

**Review by Rohit25negi — CHANGES_REQUESTED** [review]

File: custom/icds_reports/templates/icds_reports/icds_app/download.directive.html (hunk 1 above)
Comment (ShubhamBansal1997) [reply custom/icds_reports/templates/icds_reports/icds_app/download.directive.html:122]: to update the year function

File: custom/icds_reports/static/js/directives/download/download.directive.js (hunk 2 above)
Comment (ShubhamBansal1997) [reply custom/icds_reports/static/js/directives/download/download.directive.js:363]: quarter is /3 not /4

File: custom/icds_reports/templates/icds_reports/icds_app/download.directive.html (hunk 1 above)
Comment (Rohit25negi) [reply custom/icds_reports/templates/icds_reports/icds_app/download.directive.html:122]: What is being updated in onDataPeriodSelect yeah? why is model still needed?

Comment (Rohit25negi) [reply custom/icds_reports/templates/icds_reports/icds_app/download.directive.html:122]: offlined, I think we need this.

**Comment by ShubhamBansal1997** [comment]
captured (2)

**Comment by ShubhamBansal1997** [comment]
captured (2)

**Review by Rohit25negi — APPROVED** [review]

**Comment by ShubhamBansal1997** [comment]
captured (3)

**Pull request merged**